qa/tasks/vstart_runner.py

```diff
@@ -231,8 +231,23 @@ def get_file(self, path, sudo, dest_dir):
         shutil.copy(path, tmpfile)
         return tmpfile
 
+    # XXX: This method ignores the error raised when src and dst are
+    # holding same path. For teuthology, same path still represents
+    # different locations as they lie on different machines.
     def put_file(self, src, dst, sudo=False):
-        shutil.copy(src, dst)
+        if sys.version_info.major < 3:
+            exception = shutil.Error
+        elif sys.version_info.major >= 3:
+            exception = shutil.SameFileError
+
+        try:
+            shutil.copy(src, dst)
+        except exception as e:
```

Comment:
I believe it's `Exception`?

Reply:
I am not catching `Exception`. When source and destination paths are same, `shutil.copy()` in Python 2.7 raises `shutil.Error` and in Python 3 it raises `shutil.SameFileError`. So, I pick the exception to catch [here](https://github.com/ceph/ceph/pull/28793/files/3ef898e1c77e609fbadd998746b1c1126b06fc05#diff-0bfa62124ab75596476e1cfdf1a851a3R238)...

Reply:
that "here" link is broken.

I would think `Exception` would be the parent of all `shutil` exceptions.

Reply:
> that "here" link is broken.
>

Hopefully works now.

> I would think `Exception` would be the parent of all `shutil` exceptions.

Yeah, but wouldn't we want to avoid an exception as generic as that? Especially, when we can use something more precise?

Reply:
I see what you did now. What you have is fine.